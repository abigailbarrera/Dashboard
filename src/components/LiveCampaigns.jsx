import React from 'react';
import { GlobalContext } from "../context/GlobalState";

import { Table, Container, Row } from 'react-bootstrap';


export const LiveCampaigns = () => {
    const {campaigns} = React.useContext(GlobalContext);
    const liveCampaigns = campaigns.filter(campaign => campaign.status === 'Sent');

    return (
        <Container>
            <Row>
                <div className="col-12 mt-5">
                    <Table striped bordered hover>
                        <thead>
                            <tr>
                                <th>Live (sent) Campaigns</th>
                                <th>Amount Sent</th>
                                <th>Click Through Rates</th>
                            </tr>
                        </thead>
                        <tbody>
                            {liveCampaigns.map(campaign => {
                                return (
                                    <tr key={campaign.id}>
                                        <td>{campaign.name}</td>
                                        <td>{campaign.stats.sent}</td>
                                        <td>{campaign.stats.clicked}</td>
                                    </tr>
                                )
                            })}
                        </tbody>
                    </Table>
                </div>
            </Row>
        </Container>
    )
}
