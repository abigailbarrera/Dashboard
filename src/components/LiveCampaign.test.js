import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
 
import { LiveCampaigns } from './LiveCampaigns';
 
describe('LiveCampaigns', () => {
  it('snapshot renders', () => {
    let tree = create(
        <BrowserRouter>
            <LiveCampaigns />
        </BrowserRouter>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders header', () => {
    const wrapper = shallow(<LiveCampaigns />);
    const tableHeader = "Live (sent) Campaigns";

    expect(wrapper.contains(tableHeader)).toEqual(true);
  });
});
