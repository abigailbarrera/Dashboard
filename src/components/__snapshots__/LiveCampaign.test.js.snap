// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveCampaigns snapshot renders 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-12 mt-5"
    >
      <table
        className="table table-striped table-bordered table-hover"
      >
        <thead>
          <tr>
            <th>
              Live (sent) Campaigns
            </th>
            <th>
              Amount Sent
            </th>
            <th>
              Click Through Rates
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Fourth of July Promo
            </td>
            <td>
              6506
            </td>
            <td>
              6137
            </td>
          </tr>
          <tr>
            <td>
              Labor Day Sale
            </td>
            <td>
              7228
            </td>
            <td>
              6883
            </td>
          </tr>
          <tr>
            <td>
              Cotton Candy Palette Announcement
            </td>
            <td>
              7246
            </td>
            <td>
              7080
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
